File: fixtures/scale/r/small/lib/log.r
# Leveled logging to stderr.

log_write <- function(level, ...) {
  message(sprintf("[%s] %s", level, paste(..., sep = " ")))
}

log_info <- function(...) {
  log_write("INFO", ...)
}

log_warn <- function(...) {
  log_write("WARN", ...)
}

File: fixtures/scale/julia/small/src/Log.jl
module Log

export log_write, log_info, log_warn

function log_write(level::String, parts...)
    println(stderr, "[", level, "] ", join(parts, " "))
end

function log_info(parts...)
    log_write("INFO", parts...)
end

function log_warn(parts...)
    log_write("WARN", parts...)
end

end
